arch/arm/src/samv7/sam_qencoder.c

```diff
@@ -229,10 +270,15 @@ static int sam_shutdown(struct qe_lowerhalf_s *lower)
 static int sam_position(struct qe_lowerhalf_s *lower, int32_t *pos)
 {
   struct sam_lowerhalf_s *priv = (struct sam_lowerhalf_s *)lower;
+  uint32_t new_pos;
+  new_pos = sam_tc_getcounter(priv->tch);
```

Comment:
```suggestion
  uint16_t new_pos = sam_tc_getcounter(priv->tch);
```

Reply:
Gain the use of uint32_t is based on the above described overhead with 16-operations on ARM and x86.